### DISCUSSION TITLE
Android App: Add Toilet

### DISCUSSION THREAD

**Discussion by beaumoaj** [public-convenience-ltd/toiletmap#1741] [Q&A]
Hi.  I have an Andoid app that I developed to display nearby toilets using the API at https://www.toiletmap.org.uk/api.  I would like to add a function to add a new toilet to the API.  Can you direct me to some information about the API request to achieve this?  I guess I should download this repository to set up my own server to test my new functionality.  How does the authorization work for adding?  You can contact me on a.j.beaumont [at] aston.ac.uk

**Comment by ob6160** [comment]
Thanks for getting in touch @beaumoaj it's exciting that you're developing an app around our API, would love to hear more about this!
In terms of adding toilets, we offer this through our GraphQL API using a "mutation" throught the submission of a "report". Here's the relevant mutation: https://github.com/public-convenience-ltd/toiletmap/blob/main/src/api/graphql/resolvers.ts#L297-L324

We require authorisation in order for this to work. Users who login through Auth0 (https://www.toiletmap.org.uk/api/auth/login) have a session token cookie set which is then checked before allowing access. If you're able to direct users to our auth0 sign in page they should be able to have this session token issued which your application could potentially use to authorize requests once they are logged in. We may very well need to change things in order for this to work though as I'm not sure this has been done before.

Happy to discuss further over email - I'll drop you a line
